lib/rules/i18n-ellipsis.js

```diff
@@ -26,12 +38,37 @@ var rule = module.exports = function( context ) {
 
 			node.arguments.forEach( function( arg ) {
 				var argumentString = getTextContentFromNode( arg );
-				if ( argumentString && -1 !== argumentString.indexOf( '...' ) ) {
+				if ( argumentString && containsThreeDots( argumentString ) ) {
 					context.report( {
 						node: arg,
 						message: rule.ERROR_MESSAGE,
 						fix: function( fixer ) {
-							return fixer.replaceText( arg, argumentString.replace( /\.\.\./g, '…' ) );
+							if ( arg.type === 'TemplateLiteral' ) {
+								var fixes = [];
+								arg.quasis.forEach(
+									function( quasi ) {
+										if (
+											quasi.type === 'TemplateElement' &&
+											containsThreeDots( quasi.value.raw )
+										) {
+											fixes.push(
+												fixer.replaceTextRange(
+													[ quasi.start, quasi.end ],
+													replaceThreeDotsWithEllipsis( quasi.value.raw )
+												)
+											);
+										}
+									}
+								);
+								return fixes;
+							}
+							if ( arg.type === 'Literal' ) {
+								return fixer.replaceText( arg, replaceThreeDotsWithEllipsis( arg.raw ) );
+							}
+							return fixer.replaceText( arg,
+								"'" +
```

Comment:
They really don't make it easy to replace text of a BinaryExpression, huh? 😄 

That's what's being handled here, right? Should we be explicit about that? Maybe worth a `switch` on `arg.type` ?

Reply:
This is the previous implementation corrected to wrap in single quotes and escape contained single quotes.

Initially, I didn't go any further because I wasn't sure at what point to stop implementing handling for different types expressions. 

Since you've called my attention to `BinaryExpression`, I realize that that is a natural stopping point because it's the last case covered by `getTextContentFromNode`. Anything not covered by those three cases should not throw a lint error because the text won't be included.

I've changed the implementation in f0db405 so that we can recursively fix the left and right sides of `BinaryExpression`s. I'm quite satisfied with the implementation at this point.

Note that since the fixer now works "natively" with eslint functionality and doesn't rely on `getTextContentFromNode` to manually modify the string contents, the problems with quotes vanish. `+` joined strings and different quote types are untouched. Compare that to the initial implementation which mangled the source quite badly when it attempted to fix:

https://github.com/Automattic/eslint-plugin-wpcalypso/pull/45/files#diff-72e7d21b11fe78ad24318057d029a272R97